### ISSUE TITLE
Nav: Fix basic a11y issues

### ISSUE THREAD

**Issue by L-M-K-B** [grafana/grafana#80020]
There are a couple of issues with our MegaMenu when it comes to a11y. Check them and adjust if necessary.

- [x] Is a nav element, but it's not announced as such
- [x] Between hamburger menu button and the actual nav there are two other elements
- [x] When entering the MegaMenu the number of items in the main unordered list is not announced
- [x] Announce nav item level
- [x] If there are no _Starred_ items the section gets expanded but no further action is possible => _Your starred dashboards will appear here_ should be announced => item is reachable though

**Findings that were dropped after consulting with Chris Bedwell:**
- Headlines for sub-sections are not announced as headlines but only as links
- Number of list items in level 1 is announced wrongly sometimes

**Comment by grafanabot** [comment]
@L-M-K-B please add one or more appropriate labels. Here are some tips:

- if you are making an issue, TODO, or reminder for yourself or your team, please add one label that best describes the product or feature area. Please also add the issue to your project board. :rocket:

- if you are making an issue for any other reason (docs typo, you found a bug, etc), please add at least one label that best describes the product or feature that you are discussing (e.g. `area/alerting`, `datasource/loki`, `type/docs`, `type/bug`, etc). [Our issue triage](https://github.com/grafana/grafana/blob/main/ISSUE_TRIAGE.md#3-categorizing-an-issue) doc also provides additional guidance on labeling. :rocket:

 Thank you! :heart:
